src/System.Private.ServiceModel/tools/test/Bridge/WcfTestBridgeCommon/PortManager.cs

```diff
@@ -26,82 +60,111 @@ private static void RegisterForProcessExit()
                 if (!s_registeredForProcessExit)
                 {
                     AppDomain.CurrentDomain.ProcessExit += (s, e) =>
-                        {
-                            CloseAllOpenedPortsInFireWall();
-                        };
+                    {
+                        RemoveAllBridgeFirewallRules();
+                    };
                     s_registeredForProcessExit = true;
                 }
             }
         }
 
-        private static void AddNewFirewallRule(int port)
+        // Searches all existing firewall rules with the given name
+        private static INetFwRule FindRule(string name, int port)
         {
             lock (s_portLock)
             {
-                if (s_AddedRulesByPort.ContainsKey(port))
+                // Match on our special naming pattern and port
+                HashSet<string> ruleSet = new HashSet<string>();
+                foreach (var r in NetFwPolicy2.Rules)
                 {
-                    return;
+                    INetFwRule rule = (INetFwRule)r;
+                    if (String.Equals(name, rule.Name, StringComparison.OrdinalIgnoreCase) &&
+                        String.Equals(rule.LocalPorts, port.ToString(), StringComparison.Ordinal))
+                    {
+                        return rule;
+                    }
                 }
 
+                return null;
+            }
+        }
+
+        private static void AddFirewallRule(int port)
+        {
+            lock (s_portLock)
+            {
                 // If we add any rules, register to delete them at process exit.
                 RegisterForProcessExit();
 
-                // Create the FwPolicy2 object.
-                Type netFwPolicy2Type = Type.GetTypeFromProgID("HNetCfg.FwPolicy2", false);
-                INetFwPolicy2 fwPolicy2 = (INetFwPolicy2)Activator.CreateInstance(netFwPolicy2Type);
-
-                // Get the Rules object
-                INetFwRules rulesObject = fwPolicy2.Rules;
+                // If we already created this rule, we don't create it again.
+                string ruleName = String.Format("{0} {1}", s_RuleNamePrefix, port);
+                if (FindRule(ruleName, port) != null)
+                {
+                    return;
```

Comment:
I suggested just in case we wanted to give some debuggability to those who are using this script on their own. For us, I think I'm reasonably certain that our own environment wouldn't run into this. 

Yeah, I agree - forget this, we have better things to spend our time on.